### DISCUSSION TITLE
What is a generic way of getting more decimals for a datetime instead of using mysql DATETIME fsp?

### DISCUSSION THREAD

**Discussion by MathijsNL** [fastapi/sqlmodel#1219] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the SQLModel documentation, with the integrated search.
- [X] I already searched in Google "How to X in SQLModel" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to SQLModel but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to SQLModel but to [SQLAlchemy](https://github.com/sqlalchemy/sqlalchemy).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from datetime import datetime, timezone
from typing import Optional

from sqlalchemy import Column, text
from sqlalchemy.dialects import mysql
from sqlalchemy.sql import func
from sqlmodel import Field, Session, SQLModel, create_engine, select

db_user = "root"
db_pass = ""
db_host = "localhost"
db_name = "testdb"

class TestClass(SQLModel, table=True):

    __tablename__ = "testtable"

    id: Optional[int] = Field(default=None, primary_key=True)

    # datetime without decimals
    date_time: datetime = Field(default_factory=lambda: datetime.now(timezone.utc))

    # mysql DATETIME with 3 decimals
    date_time_fsp: datetime = Field(
        sa_column=Column(
            mysql.DATETIME(fsp=3), server_default=func.utc_timestamp(3), nullable=False
        )
    )


init_engine = create_engine(
    f"mysql+mysqldb://{db_user}:{db_pass}@{db_host}/", echo=True
)

with init_engine.connect() as connection:
    connection.execute(text(f"CREATE DATABASE IF NOT EXISTS {db_name}"))

engine = create_engine(f"mysql+mysqldb://root:{db_pass}@localhost/{db_name}", echo=True)

SQLModel.metadata.create_all(engine)
session = Session(engine)

session.add(TestClass())
session.commit()

with Session(engine) as session:
    statement = select(TestClass)
    results = session.exec(statement)
    for hero in results:
        print(hero)


session.close()
```

### Description

I can't figure out how to get a higher time precision in a generic way. If possible I would like to prevent dialect specific workarounds. So in short, what do I need to do differently to get a date_time column with decimals:

```
date_time: datetime = Field(default_factory=lambda: datetime.now(timezone.utc))
```

### Operating System

Linux

### SQLModel Version

0.0.22

### Python Version

Python 3.12.7

**Comment by Sammyjroberts** [comment]
bump
